src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -230,5 +282,82 @@ public void ConnectHeaders_SupplyHeaders_ToDownloadArraySizeSameAsNumberOfHeader
             // ToDownload array of the same size as the amount of headers
             Assert.Equal(headersToDownloadCount, peer2Headers.Count);
         }
+
+        /// <summary>
+        /// Issue 8 @ Create chained header tree component #1321
+        /// We have a chain and peer B is on the best chain.
+        /// After that peer C presents an alternative chain with a forkpoint below B's Tip.
+        /// After that peer A prolongs C's chain (but sends all headers from the fork point) with a few valid and one invalid.
+        /// Make sure that C's chain is not removed - only headers after that.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_MultiplePeersWithForks_CorrectTip()
```

Comment:
This test don't test what is supposed to be tested.

Test's description:
```
We have a chain and peer B is on the best chain. After that peer C presents an alternative chain with a forkpoint below B.Tip. After that peer A prolongs C's chain (but he sends all headers from the fork point) with few valid and one invalid. 
	Make sure that C's chain is not removed, only headers after that. 
```


The setup is: 

```
b1=b2=b3=b4=b5
     =c3=c4=c5=a6=a7=a8
```

After we find out that A sent invalid header a8 CHT should disconnect it and it should remove a6,a7,a8.

Instead here you are prolonging the chain normally without sending an invalid header.